common/app/common/controllers/SunriseController.java

```diff
@@ -133,8 +133,38 @@ protected RequestContext requestContext() {
         return categoryNewExtId.flatMap(extId -> categoryTree().findByExternalId(extId));
     }
 
+    private static Locale currentLocale(final String languageTag, final ProjectContext projectContext) {
+        final Locale locale = Locale.forLanguageTag(languageTag);
+        return projectContext.isLocaleAccepted(locale) ? locale : projectContext.defaultLocale();
+    }
+
+    private static CountryCode currentCountry(final Http.Session session, final ProjectContext projectContext) {
+        final String countryCodeInSession = session.get(SESSION_COUNTRY);
+        final CountryCode country = CountryCode.getByCode(countryCodeInSession, false);
+        return projectContext.isCountryAccepted(country) ? country : projectContext.defaultCountry();
+    }
+
+    private static CurrencyUnit currentCurrency(final CountryCode currentCountry, final ProjectContext projectContext) {
+        return Optional.ofNullable(currentCountry.getCurrency())
+                .map(countryCurrency -> {
+                    final CurrencyUnit currency = Monetary.getCurrency(countryCurrency.getCurrencyCode());
+                    return projectContext.isCurrencyAccepted(currency) ? currency : projectContext.defaultCurrency();
+                }).orElse(projectContext.defaultCurrency());
+    }
+
+    private static List<Locale> acceptedLocales(final String languageTag, final Http.Request request,
+                                                final ProjectContext projectContext) {
+        final ArrayList<Locale> acceptedLocales = new ArrayList<>();
+        acceptedLocales.add(currentLocale(languageTag, projectContext));
+        acceptedLocales.addAll(request.acceptLanguages().stream()
+                .map(lang -> Locale.forLanguageTag(lang.code()))
+                .collect(toList()));
+        acceptedLocales.addAll(projectContext.locales());
+        return acceptedLocales;
+    }
+
     @Nullable
-    public static String getCsrfToken(final Http.Session session) {
+    private static String getCsrfToken(final Http.Session session) {
```

Comment:
It is used. This fills the meta information. Why wouldn't it be used? I put it as private because actually it is called from the same `SunriseController`.